ts-lib/src/encryption/base58.ts

```diff
@@ -0,0 +1,63 @@
+import { HDNodeWallet } from "ethers";
+import { KeyContract } from "../keyContract";
+import { EthKeyManager } from "./interface";
+import { NearAccount } from "../types";
+import bs58 from "bs58";
+import { KeyPair } from "near-api-js";
+import { create, open } from "@nearfoundation/near-js-encryption-box";
+
+export class Base58KeyManager implements EthKeyManager {
+  // EthKeyContract connected to account for `nearPrivateKey`.
+  contract: KeyContract;
+
+  constructor(contract: KeyContract) {
+    this.contract = contract;
+  }
+
+  async encryptAndSetKey(
+    ethWallet: HDNodeWallet,
+    encryptionKey: string,
+  ): Promise<string | null> {
+    let keyPair = KeyPair.fromString(encryptionKey);
+    let encodedEthKey = this.encodeEthKey(ethWallet.privateKey);
+    const { secret: encryptedKey, nonce } = create(
+      encodedEthKey,
+      keyPair.getPublicKey().toString(),
+      encryptionKey,
+    );
+    console.log("Posting Encrypted Key", encryptedKey, nonce);
+    await this.contract.methods.set_key({ encrypted_key: encryptedKey });
+    return nonce;
+  }
+
+  async retrieveAndDecryptKey(
+    nearAccount: NearAccount,
+    nonce?: string | undefined,
```

Comment:
Why make this optional here and then use the non-null assertion on it later?

Reply:
Some Encryption libraries (like the one introduced here) use a nonce and others don't. This makes it possible to reuse the same interface for both. See #9